client/task-list/tasks/payments/generic-payment-step.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import interpolateComponents from 'interpolate-components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { Form, Link, Stepper, TextControl } from '@woocommerce/components';
+import { OPTIONS_STORE_NAME } from '@woocommerce/data';
+
+export function GenericPaymentStep( {
+	installStep,
+	markConfigured,
+	methodConfig,
+} ) {
+	const { updateOptions } = useDispatch( OPTIONS_STORE_NAME );
+	const { createNotice } = useDispatch( 'core/notices' );
+
+	const isOptionsRequesting = useSelect( ( select ) => {
+		const { isOptionsUpdating } = select( OPTIONS_STORE_NAME );
+
+		return isOptionsUpdating();
+	} );
+
+	const getInitialConfigValues = () => {
+		if ( methodConfig && methodConfig.fields ) {
+			return methodConfig.fields.reduce( ( data, field ) => {
+				return {
+					...data,
+					[ field.name ]: '',
+				};
+			}, {} );
+		}
+	};
+
+	const validate = ( values ) => {
+		if ( methodConfig && methodConfig.fields ) {
+			return methodConfig.fields.reduce( ( errors, field ) => {
+				if ( ! values[ field.name ] ) {
+					// Matches any word that is capitalized aside from abrevitions like ID.
+					const title = field.title.replace(
+						/([A-Z][a-z]+)/,
+						( val ) => val.toLowerCase()
+					);
+					return {
+						...errors,
+						[ field.name ]: __( 'Please enter your ' ) + title,
+					};
+				}
+				return errors;
+			}, {} );
+		}
+		return {};
+	};
+
+	const updateSettings = async ( values ) => {
+		// Because the GenericPaymentStep extension only works with the South African Rand
+		// currency, force the store to use it while setting the GenericPaymentStep settings
+		const options = methodConfig.getOptions
+			? methodConfig.getOptions( values )
+			: null;
+		if ( ! options ) {
+			return;
+		}
+		const update = await updateOptions( {
+			...options,
+		} );
+
+		if ( update.success ) {
+			markConfigured( methodConfig.key );
+			createNotice(
+				'success',
+				methodConfig.title +
+					__( ' connected successfully', 'woocommerce-admin' )
```

Comment:
I'm guessing this comes from the copy laid out for this but it seems a bit misleading since Paystack won't actually be connected right? but it will be "configured".

Thoughts on this wording @pmcpinto ?

Reply:
Good question. The way I see it is that in this step they're mainly connecting the plugin to a Paystack account and the configuration happens afterward in the settings. Makes sense?